src/analysis/locally_convex/with_seminorms.lean

```diff
@@ -289,6 +289,78 @@ begin
   rw (congr_fun (congr_arg (@nhds E) hp.1) 0),
   exact add_group_filter_basis.nhds_zero_has_basis _,
 end
+
+/-- The `x`-neighbourhoods of a space whose topology is induced by a family of seminorms
+are exactly the sets which contain seminorm balls around `x`.-/
+lemma with_seminorms.mem_nhds_iff (hp : with_seminorms p) (x : E) (U : set E) :
+  U ∈ nhds x ↔ ∃ (s : finset ι) (r > 0), (s.sup p).ball x r ⊆ U :=
```

Comment:
Fully agree, this is nice! I think what I had been a bit worried about was creating too many individual new lemmas and theorems so that my pull request would have contained even more small changes to review, but I guess realistically, the remedy is to just add these new changes in very small pull requests, and hope that people believe you that they will add up to something meaningful :)

Reply:
Finding a good balance between adding all the library bits to have shorter proof or making more self-contained longer proof is a subtle thing indeed, and I think everyone has a different threshold for that (I was even a bit worried that Moritz would be unhappy with this suggestion :stuck_out_tongue_closed_eyes:). Here honestly I think both would be fine anyway, but the `seminorm` API has to catch up with the much more used `normed_space` one so I think adding a lot of small useful lemmas around here is perfectly good. 

About having PRs with a lot of small little changes, again there is a balance to find to make small but not completely meaningless PRs. Again, personal opinion, but I think having medium-sized API-filling PRs is perfectly fine if the people reviewing the different changes would have been the same anyway, which is clearly the case here: splitting this into multiple PRs wouldn't have really changed anything since at this stage basically all PRs related to seminorms are reviewed by me, Moritz and Yaël.

PS: I see you added the two intermediate lemmas but didn't touch your proof of `mem_nhds_iff`. Is that on purpose?

Reply:
Thanks for the thoughts! The proof of mem_nhds_iff is now changed as well, I just got lost while committing the suggestions :)